MetaMask/metamask-extension
# Added IPFS toggle

This PR:
1. Adds an IPFS toggle
2. If IPFS Toggle is on, the user can input the IPFS value
3. In case IPFS is toggled off, no NFT image will be shown
4. Placeholder image in case, when IPFS is off, will have a show button, on click of that IPFS toggle modal pops up

* Fixes [#872](https://github.com/MetaMask/MetaMask-planning/issues/872) and [#1014](https://github.com/MetaMask/MetaMask-planning/issues/1014
)
## Screencaps


https://github.com/MetaMask/metamask-extension/assets/39872794/68eb8ee4-b3b2-47ca-bb4c-7b958d824e0a



## Manual Testing Steps

- Turn on the IPFS toggle
- Input field should be hidden
- No NFT media should be shown if it's stored in ipfs
- Toggle on, by default the default IPFS is restored, you can edit it as well

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
